- Número de grupo: 10
- Nombre de los integrantes del grupo: Gonzalo Figueroa del Val, Miguel Robledo Casal y Guillermo Calvo Suárez


# Práctica 3 : representación de conocimiento 


## Parte 1: consultas SPARQL sobre Wikidata.

### Ejemplo

Recuperar todas las instancias directas de la clase [Cabra (Q2934)](https://www.wikidata.org/wiki/Q2934) que aparecen en la base de conocimiento.

En esta práctica vamos a usar el punto de acceso [SPARQL](https://query.wikidata.org/) de Wikidata para contestar las preguntas que se formulan a continuación. Cada pregunta debe ser respondida realizando una única consulta SPARQL. Para cada una de las entidades recuperadas se mostrará __tanto su identificador como su etiqueta__ (nombre de la entidad en lenguaje natural). 

<html><head><meta charset="utf-8"></head><body><table><thead><tr><th>item</th><th>itemLabel</th></tr></thead><tbody><tr><td>http://www.wikidata.org/entity/Q151345</td><td>Billygoat Hennes</td></tr><tr><td>http://www.wikidata.org/entity/Q3569037</td><td>William Windsor</td></tr><tr><td>http://www.wikidata.org/entity/Q23003932</td><td>His Whiskers</td></tr><tr><td>http://www.wikidata.org/entity/Q24287064</td><td>Taffy</td></tr><tr><td>http://www.wikidata.org/entity/Q41239734</td><td>Lance Corporal Shenkin III</td></tr><tr><td>http://www.wikidata.org/entity/Q41240892</td><td>Lance Corporal Shenkin II</td></tr><tr><td>http://www.wikidata.org/entity/Q41241416</td><td>Lance Corporal Shenkin I</td></tr><tr><td>http://www.wikidata.org/entity/Q65326499</td><td>Konkan kanyal</td></tr></tbody></table></body></html>

El punto de acceso SPARQL de Wikidata ofrece ciertos extras sobre el estándar. Por ejemplo, la línea SERVICE de la consulta nos permite obtener automáticamente la etiqueta de cualquier item con sólo crear una nueva variable que termine en Label (por ejemplo ?itemLabel en la consulta).

Podemos utilizar el paquete [qwikidata](https://qwikidata.readthedocs.io/en/stable/) para acceder a Wikidata desde un programa Python.
#### Descomentar para instalar el paquete

In [2]:
%pip install qwikidata

Note: you may need to restart the kernel to use updated packages.


In [2]:
from qwikidata.sparql import return_sparql_query_results

sparql_query = """
SELECT ?item ?itemLabel 
WHERE 
{
  ?item wdt:P31 wd:Q2934.
  SERVICE wikibase:label { bd:serviceParam wikibase:language "en". }
}
"""
res = return_sparql_query_results(sparql_query)
res

{'head': {'vars': ['item', 'itemLabel']},
 'results': {'bindings': [{'item': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q151345'},
    'itemLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'Billygoat Hennes'}},
   {'item': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q3569037'},
    'itemLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'William Windsor'}},
   {'item': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q23003932'},
    'itemLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'His Whiskers'}},
   {'item': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q24287064'},
    'itemLabel': {'xml:lang': 'en', 'type': 'literal', 'value': 'Taffy'}},
   {'item': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q41239734'},
    'itemLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'Lance Corporal Shenkin III'}},
   {'item': {'type': 'uri',
     'value

In [3]:
# También hay alguna función de utilidad extra que nos permite, por ejemplo, obtener las subclases de una entidad.
from qwikidata.sparql import get_subclasses_of_item
Q_RIVER = "Q4022"
subclasses_of_river = get_subclasses_of_item(Q_RIVER)
subclasses_of_river

['Q4022',
 'Q100649',
 'Q159675',
 'Q543408',
 'Q591942',
 'Q653247',
 'Q794428',
 'Q1074069',
 'Q1140845',
 'Q1299258',
 'Q1460693',
 'Q1531228',
 'Q1720683',
 'Q1746748',
 'Q1986504',
 'Q2230783',
 'Q2375084',
 'Q2557367',
 'Q3073652',
 'Q3196604',
 'Q4176368',
 'Q4366834',
 'Q4392849',
 'Q4392856',
 'Q4733366',
 'Q4879406',
 'Q5371585',
 'Q7164110',
 'Q7300165',
 'Q7580767',
 'Q8026426',
 'Q10670417',
 'Q10875106',
 'Q11514609',
 'Q11553102',
 'Q11564407',
 'Q12099220',
 'Q12149663',
 'Q12309892',
 'Q12818517',
 'Q16465938',
 'Q16638716',
 'Q17105134',
 'Q18915937',
 'Q20529834',
 'Q21079327',
 'Q21573828',
 'Q25556322',
 'Q27067659',
 'Q42122394',
 'Q51205815',
 'Q55610863',
 'Q56388869',
 'Q60061427',
 'Q60823417',
 'Q71184245',
 'Q98242145',
 'Q98242280',
 'Q100717822',
 'Q10857944',
 'Q12048869',
 'Q90905898',
 'Q3058945',
 'Q21504956']

### Interfaz Linked data

Podemos recuperar la información de distintas entidades y propiedades de Wikidata y acceder a ellas mediante objetos de Python. En este ejemplo vamos a trabajar con el item [Douglas Adams (Q42)](https://www.wikidata.org/wiki/Q42).

In [4]:
from qwikidata.entity import WikidataItem, WikidataProperty, WikidataLexeme 
from qwikidata.linked_data_interface import get_entity_dict_from_api

# obtener un objeto con la información del escritor douglas Adams
Q_DOUGLAS_ADAMS = 'Q42'
q42_dict = get_entity_dict_from_api(Q_DOUGLAS_ADAMS)
q42 = WikidataItem(q42_dict)
q42

WikidataItem(label=Douglas Adams, id=Q42, description=English writer and humorist, aliases=['Douglas Noel Adams', 'Douglas Noël Adams', 'Douglas N. Adams'], enwiki_title=Douglas Adams)

In [5]:
#Podemos navegar por la información del objeto q42
q42.entity_type

'item'

In [6]:
q42.get_label()

'Douglas Adams'

In [7]:
q42.get_description()

'English writer and humorist'

In [8]:
q42.get_aliases()

['Douglas Noel Adams', 'Douglas Noël Adams', 'Douglas N. Adams']

In [9]:
# prueba a meter la info de cabra wd:Q2934
Q_CABRA = 'Q2934'
q_dict = get_entity_dict_from_api(Q_CABRA)
q = WikidataItem(q_dict)
q

WikidataItem(label=goat, id=Q2934, description=domesticated mammal raised primarily for its milk, aliases=['domestic goat'], enwiki_title=Goat)

Haz lo mismo para encontrar algo de información sobre [Steven Spielberg](https://www.wikidata.org/wiki/Q8877).

Se proponen varias consultas. Para cada una de las preguntas debes mostrar tanto la consulta como la respuesta obtenida.

 - La __consulta__ debe estar en una celda de tipo _Raw NBConvert_ para que jupyter no trate de interpretarla. Cada tripleta de la consulta debe tener un breve comentario a la derecha que la explique (los comentarios empiezan con #). 
 - La __respuesta__ debe estar en una celda de tipo _Markdown_. Puedes descargar las respuestas usando la opción _Descargar >> HTML Table_ y copiar el código HTML en esta celda. Al ejecutar la celda se mostrará en forma de tabla.
 - Si lo consideras necesario, puedes añadir celdas adicionales en formato _Markdown_ para explicar decisiones que hayas tomado al crear la consulta o cualquier otro dato que consideres interesante.
 
 
__Para resolver estas consultas necesitarás aprender algo más de SPARQL de lo que hemos contado en clase__. Los dos recursos que te recomendamos consultar son:

- [Este tutorial de SPARQL](https://www.wikidata.org/wiki/Wikidata:SPARQL_tutorial).
- [Esta recopilación de ejemplos](https://www.wikidata.org/wiki/Wikidata:SPARQL_query_service/queries/examples)


Puedes probar las consultas en el punto de acceso SPARQL[SPARQL](https://query.wikidata.org/) de Wikidata y copiar la respuesta en otra celda. 
No es necesario lanzar las consultas desde el paquete [qwikidata](https://qwikidata.readthedocs.io/en/stable/) para acceder a Wikidata desde un programa Python aunque puedes hacerlo si lo prefieres.

Lo importante son las consultas SPARQL. Cada pregunta debe ser respondida realizando una única consulta SPARQL. Para cada una de las entidades recuperadas se mostrará __tanto su identificador como su etiqueta__ (nombre de la entidad en lenguaje natural). 

### Consulta 1

[Steven Allan Spielberg (Q8877)](https://www.wikidata.org/wiki/Q8877) es uno de los directores más reconocidos y populares de la industria cinematográfica mundial. Escribe una consulta que te permita averiguar su fecha y lugar de nacimiento.

El resultado es: Nacio el 18 de diciembre de 1946	en wd:Q43196	Cincinnati
Debes escribir una consulta para obtener estos datos.
Pistas: Lugar de nacimiento (P19) de Spielberg (Q8877)
        Fecha de nacimiento (P569) de Spielberg (Q8877)

In [10]:
from qwikidata.sparql import return_sparql_query_results

sparql_query = """
SELECT ?birth_date ?birth_place ?birth_placeLabel
WHERE
{
    wd:Q8877 wdt:P569 ?birth_date.
    wd:Q8877 wdt:P19 ?birth_place.
    SERVICE wikibase:label { bd:serviceParam wikibase:language "en". }
}
"""

res = return_sparql_query_results(sparql_query)
res

{'head': {'vars': ['birth_date', 'birth_place', 'birth_placeLabel']},
 'results': {'bindings': [{'birth_date': {'datatype': 'http://www.w3.org/2001/XMLSchema#dateTime',
     'type': 'literal',
     'value': '1946-12-18T00:00:00Z'},
    'birth_place': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q43196'},
    'birth_placeLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'Cincinnati'}}]}}

<html><head><meta charset="utf-8"></head><body><table><thead><tr><th>birth_date</th><th>birth_place</th><th>birth_placeLabel</th></tr></thead><tbody><tr><td>1946-12-18T00:00:00Z</td><td>http://www.wikidata.org/entity/Q43196</td><td>Cincinnati</td></tr></tbody></table></body></html>

### Consulta 2

Además, se sabe que Spielberg es muy polifacetico.. vamos a averiguar todas las distintas profesiones (ocupaciones) (P106) de Spielberg (Q8877)en la base de conocimiento. Queremos obtener los resultados ordenados alfabéticamente por el nombre de la profesión.

In [11]:
from qwikidata.sparql import return_sparql_query_results

sparql_query = """
SELECT ?ocupaciones ?ocupacionesLabel
WHERE
{
    wd:Q8877 wdt:P106 ?ocupaciones.
    SERVICE wikibase:label { bd:serviceParam wikibase:language "en". }
}
ORDER BY ASC (?ocupacionesLabel) #Por defecto el orden se hace de forma ascendente, pero incluimos ASC para ser más concretos
"""

res = return_sparql_query_results(sparql_query)
res

{'head': {'vars': ['ocupaciones', 'ocupacionesLabel']},
 'results': {'bindings': [{'ocupaciones': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q33999'},
    'ocupacionesLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'actor'}},
   {'ocupaciones': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q10732476'},
    'ocupacionesLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'art collector'}},
   {'ocupaciones': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q43845'},
    'ocupacionesLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'businessperson'}},
   {'ocupaciones': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q3455803'},
    'ocupacionesLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'director'}},
   {'ocupaciones': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q1053574'},
    'ocupacionesLabel': {'xml:lang': 'en',
     'type': 'literal',
    

<html><head><meta charset="utf-8"></head><body><table><thead><tr><th>ocupaciones</th><th>ocupacionesLabel</th></tr></thead><tbody><tr><td>http://www.wikidata.org/entity/Q33999</td><td>actor</td></tr><tr><td>http://www.wikidata.org/entity/Q10732476</td><td>art collector</td></tr><tr><td>http://www.wikidata.org/entity/Q43845</td><td>businessperson</td></tr><tr><td>http://www.wikidata.org/entity/Q3455803</td><td>director</td></tr><tr><td>http://www.wikidata.org/entity/Q1053574</td><td>executive producer</td></tr><tr><td>http://www.wikidata.org/entity/Q10800557</td><td>film actor</td></tr><tr><td>http://www.wikidata.org/entity/Q2526255</td><td>film director</td></tr><tr><td>http://www.wikidata.org/entity/Q7042855</td><td>film editor</td></tr><tr><td>http://www.wikidata.org/entity/Q3282637</td><td>film producer</td></tr><tr><td>http://www.wikidata.org/entity/Q18844224</td><td>science fiction writer</td></tr><tr><td>http://www.wikidata.org/entity/Q28389</td><td>screenwriter</td></tr><tr><td>http://www.wikidata.org/entity/Q578109</td><td>television producer</td></tr></tbody></table></body></html>

### Consulta 3
Averigua también el nombre de su padre y de sus hermanos.


In [12]:
from qwikidata.sparql import return_sparql_query_results

sparql_query = """
SELECT ?father ?fatherLabel ?sibling ?siblingLabel
WHERE
{
    wd:Q8877 wdt:P22 ?father.
    wd:Q8877 wdt:P3373 ?sibling.
    SERVICE wikibase:label { bd:serviceParam wikibase:language "en". }
}
"""

res = return_sparql_query_results(sparql_query)
res

{'head': {'vars': ['father', 'fatherLabel', 'sibling', 'siblingLabel']},
 'results': {'bindings': [{'father': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q15784029'},
    'sibling': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q469204'},
    'fatherLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'Arnold Spielberg'},
    'siblingLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'Anne Spielberg'}}]}}

<html><head><meta charset="utf-8"></head><body><table><thead><tr><th>father</th><th>fatherLabel</th><th>sibling</th><th>siblingLabel</th></tr></thead><tbody><tr><td>http://www.wikidata.org/entity/Q15784029</td><td>Arnold Spielberg</td><td>http://www.wikidata.org/entity/Q469204</td><td>Anne Spielberg</td></tr></tbody></table></body></html>

### Consulta 4
Ahora queremos conocer el título de todas las películas que Spielberg ha dirigido. Se mostrarán ordenadas alfabéticamente y debes tener cuidado de no mostrar resultados repetidos. Ten en cuenta que puede haber distintos tipos de películas.

In [13]:
from qwikidata.sparql import return_sparql_query_results

sparql_query = """
SELECT DISTINCT ?film ?filmLabel #Utilizando DISTINCT elegimos resultados únicos
WHERE 
{
    ?film wdt:P31 wd:Q11424. #Titulo instancia (P31) de pelicula (Q11424)
    ?film wdt:P57 wd:Q8877. #Pelicula dirigida por (P57) Steven Spielberg (Q8877)
    SERVICE wikibase:label { bd:serviceParam wikibase:language "en". }
}
ORDER BY ASC (?filmLabel)
"""

res = return_sparql_query_results(sparql_query)
res

{'head': {'vars': ['film', 'filmLabel']},
 'results': {'bindings': [{'film': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q207482'},
    'filmLabel': {'xml:lang': 'en', 'type': 'literal', 'value': '1941'}},
   {'film': {'type': 'uri', 'value': 'http://www.wikidata.org/entity/Q221113'},
    'filmLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'A.I. Artificial Intelligence'}},
   {'film': {'type': 'uri', 'value': 'http://www.wikidata.org/entity/Q449743'},
    'filmLabel': {'xml:lang': 'en', 'type': 'literal', 'value': 'Always'}},
   {'film': {'type': 'uri', 'value': 'http://www.wikidata.org/entity/Q472361'},
    'filmLabel': {'xml:lang': 'en', 'type': 'literal', 'value': 'Amistad'}},
   {'film': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q18067135'},
    'filmLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'Bridge of Spies'}},
   {'film': {'type': 'uri', 'value': 'http://www.wikidata.org/entity/Q208108'},
    'filmLabe

<html><head><meta charset="utf-8"></head><body><table><thead><tr><th>film</th><th>filmLabel</th></tr></thead><tbody><tr><td>http://www.wikidata.org/entity/Q207482</td><td>1941</td></tr><tr><td>http://www.wikidata.org/entity/Q221113</td><td>A.I. Artificial Intelligence</td></tr><tr><td>http://www.wikidata.org/entity/Q449743</td><td>Always</td></tr><tr><td>http://www.wikidata.org/entity/Q472361</td><td>Amistad</td></tr><tr><td>http://www.wikidata.org/entity/Q18067135</td><td>Bridge of Spies</td></tr><tr><td>http://www.wikidata.org/entity/Q208108</td><td>Catch Me If You Can</td></tr><tr><td>http://www.wikidata.org/entity/Q320588</td><td>Close Encounters of the Third Kind</td></tr><tr><td>http://www.wikidata.org/entity/Q11621</td><td>E.T. the Extra-Terrestrial</td></tr><tr><td>http://www.wikidata.org/entity/Q271281</td><td>Empire of the Sun</td></tr><tr><td>http://www.wikidata.org/entity/Q3057871</td><td>Escape to Nowhere</td></tr><tr><td>http://www.wikidata.org/entity/Q591320</td><td>Firelight</td></tr><tr><td>http://www.wikidata.org/entity/Q182373</td><td>Indiana Jones and the Kingdom of the Crystal Skull</td></tr><tr><td>http://www.wikidata.org/entity/Q185658</td><td>Indiana Jones and the Last Crusade</td></tr><tr><td>http://www.wikidata.org/entity/Q179215</td><td>Indiana Jones and the Temple of Doom</td></tr><tr><td>http://www.wikidata.org/entity/Q189505</td><td>Jaws</td></tr><tr><td>http://www.wikidata.org/entity/Q167726</td><td>Jurassic Park</td></tr><tr><td>http://www.wikidata.org/entity/Q18276472</td><td>Jurassic Park 3D</td></tr><tr><td>http://www.wikidata.org/entity/Q32433</td><td>Lincoln</td></tr><tr><td>http://www.wikidata.org/entity/Q244604</td><td>Minority Report</td></tr><tr><td>http://www.wikidata.org/entity/Q152456</td><td>Munich</td></tr><tr><td>http://www.wikidata.org/entity/Q174284</td><td>Raiders of the Lost Ark</td></tr><tr><td>http://www.wikidata.org/entity/Q165817</td><td>Saving Private Ryan</td></tr><tr><td>http://www.wikidata.org/entity/Q483941</td><td>Schindler's List</td></tr><tr><td>http://www.wikidata.org/entity/Q223299</td><td>The Color Purple</td></tr><tr><td>http://www.wikidata.org/entity/Q200873</td><td>The Lost World: Jurassic Park</td></tr><tr><td>http://www.wikidata.org/entity/Q30203425</td><td>The Post</td></tr><tr><td>http://www.wikidata.org/entity/Q432526</td><td>The Sugarland Express</td></tr><tr><td>http://www.wikidata.org/entity/Q318766</td><td>The Terminal</td></tr><tr><td>http://www.wikidata.org/entity/Q11791805</td><td>The Unfinished Journey</td></tr><tr><td>http://www.wikidata.org/entity/Q1330737</td><td>Twilight Zone: The Movie</td></tr><tr><td>http://www.wikidata.org/entity/Q218589</td><td>War Horse</td></tr><tr><td>http://www.wikidata.org/entity/Q202028</td><td>War of the Worlds</td></tr><tr><td>http://www.wikidata.org/entity/Q63643994</td><td>West Side Story</td></tr></tbody></table></body></html>

### Consulta 5

Ahora queremos conocer el título de todas las películas que Spielberg ha dirigido. Se mostrarán ordenadas alfabéticamente y debes tener cuidado de no mostrar resultados repetidos. Ten en cuenta que puede haber distintos tipos de películas.

### ¡¡¡La consulta 5 es igual que la 4!!!

### Consulta 6

Spielberg es sin duda un director prolífico. ¿Exactamente cuántas películas de ciencia ficción ha dirigido?

In [15]:
from qwikidata.sparql import return_sparql_query_results

sparql_query = """
SELECT (COUNT(*) as ?count)
WHERE 
{
    ?film wdt:P31 wd:Q11424. #Id pelicula instancia (P31) de pelicula (Q11424)
    ?film wdt:P57 wd:Q8877. #Pelicula dirigida por (P57) Steven Spielberg (Q8877)
    ?film wdt:P136 wd:Q471839. #Pelicula de genero (P136) science fiction film (Q471839)
    SERVICE wikibase:label { bd:serviceParam wikibase:language "en". }
}
"""

res = return_sparql_query_results(sparql_query)
res

{'head': {'vars': ['count']},
 'results': {'bindings': [{'count': {'datatype': 'http://www.w3.org/2001/XMLSchema#integer',
     'type': 'literal',
     'value': '9'}}]}}

<html><head><meta charset="utf-8"></head><body><table><thead><tr><th>count</th></tr></thead><tbody><tr><td>9</td></tr></tbody></table></body></html>

### Consulta 7

Es importante que las películas tengan una duración adecuada, ni muy cortas ni demasiado largas. De todas las películas que ha dirigido Spielberg, ¿cuales duran entre 90 y 150 minutos? Para cada película muestra el título y la duración. Los resultados se deben mostrar ordenados alfabéticamente.

In [14]:
from qwikidata.sparql import return_sparql_query_results

sparql_query = """
SELECT ?film ?filmLabel ?duracion
WHERE 
{
    ?film wdt:P31 wd:Q11424. #Titulo instancia (P31) de pelicula (Q11424)
    ?film wdt:P57 wd:Q8877. #Pelicula dirigida por (P57) Steven Spielberg (Q8877)
    ?film wdt:P2047 ?duracion #Duración de un evento (en este caso pelicula) --> P2047
    FILTER(?duracion > 90 && ?duracion < 150) #Filtrado
    SERVICE wikibase:label { bd:serviceParam wikibase:language "en". }
}
ORDER BY ASC (?filmLabel)
"""

res = return_sparql_query_results(sparql_query)
res

{'head': {'vars': ['film', 'filmLabel', 'duracion']},
 'results': {'bindings': [{'film': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q207482'},
    'duracion': {'datatype': 'http://www.w3.org/2001/XMLSchema#decimal',
     'type': 'literal',
     'value': '113'},
    'filmLabel': {'xml:lang': 'en', 'type': 'literal', 'value': '1941'}},
   {'film': {'type': 'uri', 'value': 'http://www.wikidata.org/entity/Q221113'},
    'duracion': {'datatype': 'http://www.w3.org/2001/XMLSchema#decimal',
     'type': 'literal',
     'value': '146'},
    'filmLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'A.I. Artificial Intelligence'}},
   {'film': {'type': 'uri', 'value': 'http://www.wikidata.org/entity/Q449743'},
    'duracion': {'datatype': 'http://www.w3.org/2001/XMLSchema#decimal',
     'type': 'literal',
     'value': '117'},
    'filmLabel': {'xml:lang': 'en', 'type': 'literal', 'value': 'Always'}},
   {'film': {'type': 'uri',
     'value': 'http://www.wikidata.

<html><head><meta charset="utf-8"></head><body><table><thead><tr><th>film</th><th>filmLabel</th><th>duracion</th></tr></thead><tbody><tr><td>http://www.wikidata.org/entity/Q207482</td><td>1941</td><td>113</td></tr><tr><td>http://www.wikidata.org/entity/Q221113</td><td>A.I. Artificial Intelligence</td><td>146</td></tr><tr><td>http://www.wikidata.org/entity/Q449743</td><td>Always</td><td>117</td></tr><tr><td>http://www.wikidata.org/entity/Q18067135</td><td>Bridge of Spies</td><td>142</td></tr><tr><td>http://www.wikidata.org/entity/Q208108</td><td>Catch Me If You Can</td><td>135</td></tr><tr><td>http://www.wikidata.org/entity/Q320588</td><td>Close Encounters of the Third Kind</td><td>134</td></tr><tr><td>http://www.wikidata.org/entity/Q11621</td><td>E.T. the Extra-Terrestrial</td><td>115</td></tr><tr><td>http://www.wikidata.org/entity/Q591320</td><td>Firelight</td><td>135</td></tr><tr><td>http://www.wikidata.org/entity/Q182373</td><td>Indiana Jones and the Kingdom of the Crystal Skull</td><td>123</td></tr><tr><td>http://www.wikidata.org/entity/Q185658</td><td>Indiana Jones and the Last Crusade</td><td>122</td></tr><tr><td>http://www.wikidata.org/entity/Q179215</td><td>Indiana Jones and the Temple of Doom</td><td>114</td></tr><tr><td>http://www.wikidata.org/entity/Q189505</td><td>Jaws</td><td>124</td></tr><tr><td>http://www.wikidata.org/entity/Q167726</td><td>Jurassic Park</td><td>123</td></tr><tr><td>http://www.wikidata.org/entity/Q18276472</td><td>Jurassic Park 3D</td><td>126</td></tr><tr><td>http://www.wikidata.org/entity/Q244604</td><td>Minority Report</td><td>145</td></tr><tr><td>http://www.wikidata.org/entity/Q174284</td><td>Raiders of the Lost Ark</td><td>111</td></tr><tr><td>http://www.wikidata.org/entity/Q200873</td><td>The Lost World: Jurassic Park</td><td>129</td></tr><tr><td>http://www.wikidata.org/entity/Q30203425</td><td>The Post</td><td>115</td></tr><tr><td>http://www.wikidata.org/entity/Q432526</td><td>The Sugarland Express</td><td>106</td></tr><tr><td>http://www.wikidata.org/entity/Q318766</td><td>The Terminal</td><td>124</td></tr><tr><td>http://www.wikidata.org/entity/Q1330737</td><td>Twilight Zone: The Movie</td><td>101</td></tr><tr><td>http://www.wikidata.org/entity/Q218589</td><td>War Horse</td><td>146</td></tr><tr><td>http://www.wikidata.org/entity/Q202028</td><td>War of the Worlds</td><td>116</td></tr></tbody></table></body></html>

##  Parte II: 
Elige un tema de tu interés y realiza consultas para obtener información relevante de la base de conocimiento de Wikidata. 
Por ejemplo, puede ser una serie de televisión (como Game of Thrones (Q23572) o los  Bridgerton (Q85748936), saga de películas (Star Wars (Q462)) o tu cantante o grupo favorito (Alejandro Sanz (Q276170)). 
O cualquier otro tema: instrumentos musicales, videojuegos (Q659563), geografía, arte,.. 


# The Rolling Stones

Se han realizado 7 consultas sobre el grupo "The Rolling Stones", comenzando con 3 consultas básicas y terminando con 5 consultas más complejas que abarcan prácticamente toda la información relevante del grupo

### Consulta 1

Información básica sobre el grupo, tipo de grupo, año en el que se formó, localidad y país 

In [17]:
from qwikidata.sparql import return_sparql_query_results

sparql_query = """
SELECT ?tipo ?tipoLabel (YEAR(?anio) as ?fundacion) ?place ?placeLabel ?pais ?paisLabel
WHERE
{
    wd:Q11036 wdt:P31 ?tipo.
    wd:Q11036 wdt:P571 ?anio.
    wd:Q11036 wdt:P740 ?place.
    wd:Q11036 wdt:P17 ?pais.
    SERVICE wikibase:label { bd:serviceParam wikibase:language "en". }
}
"""
res = return_sparql_query_results(sparql_query)
res

{'head': {'vars': ['tipo',
   'tipoLabel',
   'fundacion',
   'place',
   'placeLabel',
   'pais',
   'paisLabel']},
 'results': {'bindings': [{'tipo': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q5741069'},
    'place': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q858775'},
    'pais': {'type': 'uri', 'value': 'http://www.wikidata.org/entity/Q145'},
    'tipoLabel': {'xml:lang': 'en', 'type': 'literal', 'value': 'rock group'},
    'placeLabel': {'xml:lang': 'en', 'type': 'literal', 'value': 'Dartford'},
    'paisLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'United Kingdom'},
    'fundacion': {'datatype': 'http://www.w3.org/2001/XMLSchema#integer',
     'type': 'literal',
     'value': '1962'}}]}}

<html><head><meta charset="utf-8"></head><body><table><thead><tr><th>tipo</th><th>tipoLabel</th><th>fundacion</th><th>place</th><th>placeLabel</th><th>pais</th><th>paisLabel</th></tr></thead><tbody><tr><td>http://www.wikidata.org/entity/Q5741069</td><td>rock group</td><td>1962</td><td>http://www.wikidata.org/entity/Q858775</td><td>Dartford</td><td>http://www.wikidata.org/entity/Q145</td><td>United Kingdom</td></tr></tbody></table></body></html>

### Consulta 2

Nombres de las personas que han formado parte del grupo y una foto de cada uno de ellos

In [16]:
from qwikidata.sparql import return_sparql_query_results

sparql_query = """
SELECT ?miembros ?miembrosLabel ?image
WHERE
{
    wd:Q11036 wdt:P527 ?miembros.
    ?miembros wdt:P18 ?image
    SERVICE wikibase:label { bd:serviceParam wikibase:language "en". }
}
"""
res = return_sparql_query_results(sparql_query)
res

{'head': {'vars': ['miembros', 'miembrosLabel', 'image']},
 'results': {'bindings': [{'miembros': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q128121'},
    'image': {'type': 'uri',
     'value': 'http://commons.wikimedia.org/wiki/Special:FilePath/Rolling%20Stones%2004.jpg'},
    'miembrosLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'Mick Jagger'}},
   {'miembros': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q189599'},
    'image': {'type': 'uri',
     'value': 'http://commons.wikimedia.org/wiki/Special:FilePath/Keith%20Richards%202%20Berlinale%202008.jpg'},
    'miembrosLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'Keith Richards'}},
   {'miembros': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q204943'},
    'image': {'type': 'uri',
     'value': 'http://commons.wikimedia.org/wiki/Special:FilePath/Brian%20Jones%2C%20Statesboro%2C%20Georgia%2C%20May%204%2C%201965%20%28377872218%29.jpg'},
    'm

<html><head><meta charset="utf-8"></head><body><table><thead><tr><th>miembros</th><th>miembrosLabel</th><th>image</th></tr></thead><tbody><tr><td>http://www.wikidata.org/entity/Q128121</td><td>Mick Jagger</td><td>http://commons.wikimedia.org/wiki/Special:FilePath/Rolling%20Stones%2004.jpg</td></tr><tr><td>http://www.wikidata.org/entity/Q189599</td><td>Keith Richards</td><td>http://commons.wikimedia.org/wiki/Special:FilePath/Keith%20Richards%202%20Berlinale%202008.jpg</td></tr><tr><td>http://www.wikidata.org/entity/Q204943</td><td>Brian Jones</td><td>http://commons.wikimedia.org/wiki/Special:FilePath/Brian%20Jones%2C%20Statesboro%2C%20Georgia%2C%20May%204%2C%201965%20%28377872218%29.jpg</td></tr><tr><td>http://www.wikidata.org/entity/Q211094</td><td>Ronnie Wood</td><td>http://commons.wikimedia.org/wiki/Special:FilePath/Ronnie%20Wood%20%2855208398%29.jpeg</td></tr><tr><td>http://www.wikidata.org/entity/Q216051</td><td>Charlie Watts</td><td>http://commons.wikimedia.org/wiki/Special:FilePath/Charlie%20Watts%20Hannover%2019-07-2006.jpg</td></tr><tr><td>http://www.wikidata.org/entity/Q312559</td><td>Bill Wyman</td><td>http://commons.wikimedia.org/wiki/Special:FilePath/Bill%20Wyman%202009.jpg</td></tr><tr><td>http://www.wikidata.org/entity/Q312696</td><td>Mick Taylor</td><td>http://commons.wikimedia.org/wiki/Special:FilePath/Mick%20Taylor%20onstage%20The%20Lucky.jpg</td></tr></tbody></table></body></html>

### Consulta 3

Géneros a los que pertenece el grupo ordenados alfabéticamente

In [21]:
from qwikidata.sparql import return_sparql_query_results

sparql_query = """
SELECT ?genre ?genreLabel
WHERE
{
    wd:Q11036 wdt:P136 ?genre.
    SERVICE wikibase:label { bd:serviceParam wikibase:language "en". }
}
ORDER BY ASC (?genreLabel)
"""
res = return_sparql_query_results(sparql_query)
res

{'head': {'vars': ['genre', 'genreLabel']},
 'results': {'bindings': [{'genre': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q9759'},
    'genreLabel': {'xml:lang': 'en', 'type': 'literal', 'value': 'blues'}},
   {'genre': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q193355'},
    'genreLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'blues rock'}},
   {'genre': {'type': 'uri', 'value': 'http://www.wikidata.org/entity/Q83440'},
    'genreLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'country music'}},
   {'genre': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q613408'},
    'genreLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'country rock'}},
   {'genre': {'type': 'uri', 'value': 'http://www.wikidata.org/entity/Q83270'},
    'genreLabel': {'xml:lang': 'en', 'type': 'literal', 'value': 'hard rock'}},
   {'genre': {'type': 'uri', 'value': 'http://www.wikidata.org/entity/Q27190'},
   

<html><head><meta charset="utf-8"></head><body><table><thead><tr><th>genre</th><th>genreLabel</th></tr></thead><tbody><tr><td>http://www.wikidata.org/entity/Q9759</td><td>blues</td></tr><tr><td>http://www.wikidata.org/entity/Q193355</td><td>blues rock</td></tr><tr><td>http://www.wikidata.org/entity/Q83440</td><td>country music</td></tr><tr><td>http://www.wikidata.org/entity/Q613408</td><td>country rock</td></tr><tr><td>http://www.wikidata.org/entity/Q83270</td><td>hard rock</td></tr><tr><td>http://www.wikidata.org/entity/Q27190</td><td>protopunk</td></tr><tr><td>http://www.wikidata.org/entity/Q206159</td><td>psychedelic rock</td></tr><tr><td>http://www.wikidata.org/entity/Q45981</td><td>rhythm and blues</td></tr><tr><td>http://www.wikidata.org/entity/Q7749</td><td>rock and roll</td></tr><tr><td>http://www.wikidata.org/entity/Q11399</td><td>rock music</td></tr></tbody></table></body></html>

### Consulta 4

Álbumes interpretados por el grupo ordenados alfabéticamente (en el resultado aparecen repetidos pero también tienen identificador repetido y el distinct no funciona para eso)

In [22]:
from qwikidata.sparql import return_sparql_query_results

sparql_query = """
SELECT DISTINCT ?album ?albumLabel
WHERE 
{
    ?album wdt:P31 wd:Q208569. #Titulo instancia (P31) de álbum de estudio (Q208569)
    ?album wdt:P175 wd:Q11036. #Album interpretado por (P175) The Rolling Stones (Q11036)
    SERVICE wikibase:label { bd:serviceParam wikibase:language "en". }
}
ORDER BY(?albumLabel)
"""
res = return_sparql_query_results(sparql_query)
res

{'head': {'vars': ['album', 'albumLabel']},
 'results': {'bindings': [{'album': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q175070'},
    'albumLabel': {'xml:lang': 'en', 'type': 'literal', 'value': '12 X 5'}},
   {'album': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q389281'},
    'albumLabel': {'xml:lang': 'en', 'type': 'literal', 'value': 'Aftermath'}},
   {'album': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q62579599'},
    'albumLabel': {'xml:lang': 'en', 'type': 'literal', 'value': 'Aftermath'}},
   {'album': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q63107624'},
    'albumLabel': {'xml:lang': 'en', 'type': 'literal', 'value': 'Aftermath'}},
   {'album': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q1129432'},
    'albumLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': "December's Children (And Everybody's)"}},
   {'album': {'type': 'uri',
     'value': 'http://www.wikidata

<html><head><meta charset="utf-8"></head><body><table><thead><tr><th>album</th><th>albumLabel</th></tr></thead><tbody><tr><td>http://www.wikidata.org/entity/Q175070</td><td>12 X 5</td></tr><tr><td>http://www.wikidata.org/entity/Q62579599</td><td>Aftermath</td></tr><tr><td>http://www.wikidata.org/entity/Q63107624</td><td>Aftermath</td></tr><tr><td>http://www.wikidata.org/entity/Q389281</td><td>Aftermath</td></tr><tr><td>http://www.wikidata.org/entity/Q1129432</td><td>December's Children (And Everybody's)</td></tr><tr><td>http://www.wikidata.org/entity/Q3054276</td><td>England's Newest Hit Makers</td></tr><tr><td>http://www.wikidata.org/entity/Q379339</td><td>Let It Bleed</td></tr><tr><td>http://www.wikidata.org/entity/Q62579329</td><td>Out of Our Heads</td></tr><tr><td>http://www.wikidata.org/entity/Q580504</td><td>Out of Our Heads</td></tr><tr><td>http://www.wikidata.org/entity/Q745850</td><td>Sticky Fingers</td></tr><tr><td>http://www.wikidata.org/entity/Q547315</td><td>Tattoo You</td></tr><tr><td>http://www.wikidata.org/entity/Q591855</td><td>The Rolling Stones</td></tr><tr><td>http://www.wikidata.org/entity/Q930757</td><td>The Rolling Stones No. 2</td></tr></tbody></table></body></html>

### Consulta 5

Lista de canciones pertenecientes al álbum "Aftermath" ordenadas alfabéticamente (hay que tener ojo, algunas canciones son instancia de "song" y otras son instancia de "single" y por eso utilizamos VALUES). No muestra todas las canciones del álbum realmente puesto que alguna de ellas en wikidata no detalla a que álbum pertenece.

In [20]:
from qwikidata.sparql import return_sparql_query_results

sparql_query = """
SELECT DISTINCT ?track ?trackLabel
WHERE 
{
    VALUES ?canciones {wd:Q134556 wd:Q7366}
    ?track wdt:P31 ?canciones. #Titulo instancia (P31) de álbum de estudio (Q208569)
    ?track wdt:P361 wd:Q389281. #Album interpretado por (P175) The Rolling Stones (Q11036)
    SERVICE wikibase:label { bd:serviceParam wikibase:language "en". }
}
ORDER BY(?trackLabel)
"""
res = return_sparql_query_results(sparql_query)
res

{'head': {'vars': ['track', 'trackLabel']},
 'results': {'bindings': [{'track': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q10506995'},
    'trackLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': "Goin' Home"}},
   {'track': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q28434526'},
    'trackLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'I Am Waiting'}},
   {'track': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q2553566'},
    'trackLabel': {'xml:lang': 'en', 'type': 'literal', 'value': 'Lady Jane'}},
   {'track': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q1259099'},
    'trackLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': "Mother's Little Helper"}},
   {'track': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q2432166'},
    'trackLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'Out of Time'}},
   {'track': {'type': 'uri',
     'va

<html><head><meta charset="utf-8"></head><body><table><thead><tr><th>track</th><th>trackLabel</th></tr></thead><tbody><tr><td>http://www.wikidata.org/entity/Q10506995</td><td>Goin' Home</td></tr><tr><td>http://www.wikidata.org/entity/Q28434526</td><td>I Am Waiting</td></tr><tr><td>http://www.wikidata.org/entity/Q2553566</td><td>Lady Jane</td></tr><tr><td>http://www.wikidata.org/entity/Q1259099</td><td>Mother's Little Helper</td></tr><tr><td>http://www.wikidata.org/entity/Q2432166</td><td>Out of Time</td></tr><tr><td>http://www.wikidata.org/entity/Q1046717</td><td>Paint It Black</td></tr><tr><td>http://www.wikidata.org/entity/Q3646713</td><td>Stupid Girl</td></tr></tbody></table></body></html>

### Consulta 6

Premios conseguidos por el grupo y fecha ordenados cronológicamente por año de la ceremonia.

In [11]:
from qwikidata.sparql import return_sparql_query_results

sparql_query = """
SELECT ?award ?awardLabel (YEAR(?fecha) as ?anio)
WHERE 
{
  wd:Q11036 p:P166 ?nodo.
  ?nodo ps:P166 ?award.
  ?nodo pq:P585 ?fecha.
  SERVICE wikibase:label { bd:serviceParam wikibase:language "en". }
}
ORDER BY ASC (?fecha)
"""
res = return_sparql_query_results(sparql_query)
res

{'head': {'vars': ['award', 'awardLabel', 'anio']},
 'results': {'bindings': [{'award': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q935843'},
    'awardLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'Grammy Lifetime Achievement Award'},
    'anio': {'datatype': 'http://www.w3.org/2001/XMLSchema#integer',
     'type': 'literal',
     'value': '1985'}},
   {'award': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q935843'},
    'awardLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'Grammy Lifetime Achievement Award'},
    'anio': {'datatype': 'http://www.w3.org/2001/XMLSchema#integer',
     'type': 'literal',
     'value': '1986'}},
   {'award': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q179191'},
    'awardLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'Rock and Roll Hall of Fame'},
    'anio': {'datatype': 'http://www.w3.org/2001/XMLSchema#integer',
     'type': 'literal',
     'va

<html><head><meta charset="utf-8"></head><body><table><thead><tr><th>award</th><th>awardLabel</th><th>anio</th></tr></thead><tbody><tr><td>http://www.wikidata.org/entity/Q935843</td><td>Grammy Lifetime Achievement Award</td><td>1985</td></tr><tr><td>http://www.wikidata.org/entity/Q935843</td><td>Grammy Lifetime Achievement Award</td><td>1986</td></tr><tr><td>http://www.wikidata.org/entity/Q179191</td><td>Rock and Roll Hall of Fame</td><td>1989</td></tr><tr><td>http://www.wikidata.org/entity/Q16963163</td><td>Juno Award for International Entertainer of the Year</td><td>1991</td></tr><tr><td>http://www.wikidata.org/entity/Q821861</td><td>Michael Jackson Video Vanguard Award</td><td>1994</td></tr><tr><td>http://www.wikidata.org/entity/Q691892</td><td>Grammy Award for Best Rock Album</td><td>1994</td></tr><tr><td>http://www.wikidata.org/entity/Q2976730</td><td>Grammy Award for Best Music Video</td><td>1994</td></tr><tr><td>http://www.wikidata.org/entity/Q615682</td><td>MTV Video Music Award for Best Cinematography</td><td>1995</td></tr><tr><td>http://www.wikidata.org/entity/Q372123</td><td>MTV Video Music Award for Best Visual Effects</td><td>1995</td></tr><tr><td>http://www.wikidata.org/entity/Q81180</td><td>Grammy Hall of Fame</td><td>1997</td></tr><tr><td>http://www.wikidata.org/entity/Q81180</td><td>Grammy Hall of Fame</td><td>1998</td></tr><tr><td>http://www.wikidata.org/entity/Q81180</td><td>Grammy Hall of Fame</td><td>1998</td></tr><tr><td>http://www.wikidata.org/entity/Q81180</td><td>Grammy Hall of Fame</td><td>2004</td></tr><tr><td>http://www.wikidata.org/entity/Q81180</td><td>Grammy Hall of Fame</td><td>2011</td></tr><tr><td>http://www.wikidata.org/entity/Q81180</td><td>Grammy Hall of Fame</td><td>2013</td></tr></tbody></table></body></html>

### Consulta 7

Canciones más populares del grupo, álbum al que pertenecen y año del álbum, ordenados por orden cronológico de publicación del álbum.

In [10]:
from qwikidata.sparql import return_sparql_query_results

sparql_query = """
SELECT ?track ?trackLabel ?name ?nameLabel (YEAR(?fecha) as ?anio)
WHERE 
{
  wd:Q11036 p:P800 ?nodo.
  ?nodo ps:P800 ?track.
  ?track p:P361 ?album.
  ?album ps:P361 ?name.
  ?name wdt:P577 ?fecha.
  SERVICE wikibase:label { bd:serviceParam wikibase:language "en". }
}
ORDER BY ASC (?fecha)
"""
res = return_sparql_query_results(sparql_query)
res

{'head': {'vars': ['track', 'trackLabel', 'name', 'nameLabel', 'anio']},
 'results': {'bindings': [{'track': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q158553'},
    'name': {'type': 'uri', 'value': 'http://www.wikidata.org/entity/Q580504'},
    'trackLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': "(I Can't Get No) Satisfaction"},
    'nameLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'Out of Our Heads'},
    'anio': {'datatype': 'http://www.w3.org/2001/XMLSchema#integer',
     'type': 'literal',
     'value': '1965'}},
   {'track': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q2071495'},
    'name': {'type': 'uri',
     'value': 'http://www.wikidata.org/entity/Q1129432'},
    'trackLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': 'Get Off of My Cloud'},
    'nameLabel': {'xml:lang': 'en',
     'type': 'literal',
     'value': "December's Children (And Everybody's)"},
    'anio': {'datatype': 'htt

<html><head><meta charset="utf-8"></head><body><table><thead><tr><th>track</th><th>trackLabel</th><th>name</th><th>nameLabel</th><th>anio</th></tr></thead><tbody><tr><td>http://www.wikidata.org/entity/Q158553</td><td>(I Can't Get No) Satisfaction</td><td>http://www.wikidata.org/entity/Q580504</td><td>Out of Our Heads</td><td>1965</td></tr><tr><td>http://www.wikidata.org/entity/Q2071495</td><td>Get Off of My Cloud</td><td>http://www.wikidata.org/entity/Q1129432</td><td>December's Children (And Everybody's)</td><td>1965</td></tr><tr><td>http://www.wikidata.org/entity/Q1046717</td><td>Paint It Black</td><td>http://www.wikidata.org/entity/Q389281</td><td>Aftermath</td><td>1966</td></tr><tr><td>http://www.wikidata.org/entity/Q2418659</td><td>Honky Tonk Women</td><td>http://www.wikidata.org/entity/Q27568</td><td>Through the Past, Darkly (Big Hits Vol. 2)</td><td>1969</td></tr><tr><td>http://www.wikidata.org/entity/Q2312462</td><td>Miss You</td><td>http://www.wikidata.org/entity/Q945740</td><td>Some Girls</td><td>1978</td></tr><tr><td>http://www.wikidata.org/entity/Q1477141</td><td>Jumpin' Jack Flash</td><td>http://www.wikidata.org/entity/Q3622215</td><td>Aretha</td><td>1986</td></tr></tbody></table></body></html>